### ISSUE TITLE
Legibility / contrast warning is displayed in wrong situations

### ISSUE THREAD

**Issue by draganescu** [WordPress/gutenberg#14687]
**Describe the bug**
When a Paragraph block is added and from colour settings we pick a dark colour, we can see the contrast warning even if the contrast is fine. If  we also pick a light text colour the warning disappears.

At the same time, when a Paragraph block is added and from colour settings we pick a light text colour the poor contrast warning doesn't show unless we also pick a light bg colour.

The problem in sum seems to be that for the contrast warning to work properly both the bg and the text colours must be picked.

**To Reproduce**
1. Make a post
2. Add a Paragraph block
3. Paste some text inside
4. Pick a dark blue Background color
5. Notice the contrast warning.

**Expected behavior**
The contrast warning work as the eye does, when we see light text on light background, not depending on what properties the block has selected.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: OSX
 - Browser Chrome
 - Version latest

**Additional context**
- This has been tested on the current Gutenberg master, and discovered by @afercia

**Comment by gziolo** [comment]
Discussed on WordPress.org Slack (requires registration) as well:
https://wordpress.slack.com/archives/C02QB2JS7/p1553781179021700

**Comment by Jackie6** [comment]
Hi @draganescu I do not understand why we need the following lines of code 
https://github.com/WordPress/gutenberg/blob/359858da0675943d8a759a0a7c03e7b3846536f5/packages/block-library/src/paragraph/edit.js#L228-L230
Specifically, if we have these lines, when we change the background color of the block to dark blue (secondary color), the text color will be automatically changed to white, I do not understand why we should change the text color.
![qkoUUNG7Tj](https://user-images.githubusercontent.com/24849874/55203208-12e39500-51a1-11e9-9a18-f03cc01a3839.gif)

**Comment by draganescu** [comment]
@Jackie6 It is a nice feature that helps with making sure we have default accessibility ON, that is it prevents mistakes of default black text on dark blue background.

**Comment by Jackie6** [comment]
I agree it is a nice feature. But I do not find any documentation about this feature. Do you know it? Or maybe we could write a doc for this feature. Because I think this design especially these lines of code is a little confusing at first glance.

**Comment by karmatosed** [comment]
I am removing the 'User Experience (UX)' label as part of a label cleanup. It's not being used anymore consistently so let's try and keep to 'needs design' and 'needs design feedback'. If we find a need for another label we can consider it but having those 2 should cover this.

**Comment by mapk** [comment]
After reviewing this issue in the design slack triage today, I'm noticing a number of bugs. I'll add some screenshots to help visualize the problems.

1. The contrast warning msg is incorrect. Making the text darker will not fix this issue.

![Screen Shot 2019-10-22 at 9 46 12 AM](https://user-images.githubusercontent.com/617986/67309829-67d65800-f4b1-11e9-8707-c40fc2f8fd40.png)

2. There is no contrast warning when I have light text on a light background.

![Screen Shot 2019-10-22 at 9 48 43 AM](https://user-images.githubusercontent.com/617986/67309867-74f34700-f4b1-11e9-9b08-43a625700cda.png)

**Comment by mapk** [comment]
Looks to me like the logic behind when to display a contrast warning needs some help. And some of the wording in the messages needs refining. Do we have a list of the messages anywhere so I can review them all, @draganescu ?

**Comment by draganescu** [comment]
Tested again and with the new color controls on paragraphs and other blocks (also tested the cover block) this does not seem to be a problem anymore.
